import sys
import traceback
sys.path.append('..')

from redteam import redteam

r = redteam.RedTeam()
edb = r.EDB
trello = r.RedTeamTrello

test_system = 'RHEL 7.0, no patches'

try:
    exploits = trello.parse_exploits('rhel7_exploits.csv')
    counter = 0
    title = ''
    for edbid in exploits.keys():
        exploit = exploits[edbid]
        if exploit.get('curated'):
            title = edb.get_title(edbid)
            list_id = trello.get_curated_list_id()
            card_exists = trello.get_card_id(title, list_id=list_id)
            if card_exists:
                continue
            else:
                description = trello.render_description('curated', exploit)
                trello.add_card(list_id, title, description)
        else:
            title = edb.get_title(exploit['edb_id'])
            list_id = trello.get_mapped_list_id()
            card_exists = trello.get_card_id(title)
            if card_exists:
                continue
            else:
                description = trello.render_description('mapped', exploit)
                trello.add_card(list_id, title, description)
        print('created card ' + str(counter) + ' ' + title)
        counter += 1

except Exception as e:
    # print('crowdsource_exploits caught exception: ' + str(e))
    traceback.print_exc()
1